### PR TITLE
Fix serializer block with anchors, remove trailing slash before the anchor

### PR DESCRIPTION
**Pull request by sneridagh**
@davisagli we overlooked this one :( the roundtrip was bugged because of the trailing slash. Please take a look at the regex, if it's fine like this.

### CODE REVIEW

**Review by tisto — APPROVED** [review]
LGTM. However I'd like @davisagli to have a second look at it.

File: news/1748.bugfix [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fix blocks resolveuid serializer with suffixes, remove trailing slash added before anchors
+[sneridagh]
```
Comment (davisagli) [inline news/1748.bugfix:2]:
```suggestion
Fixed an edge case in the blocks resolveuid transforms with a trailing slash before a fragment. @sneridagh 
```

**Comment by davisagli** [comment]
I added a related fix for the deserializer too. Checking what jenkins says...

**Review by davisagli — APPROVED** [review]

**Comment by tisto** [comment]
@davisagli @sneridagh released with https://pypi.org/project/plone.restapi/9.4.2/#section-1.

**Pull request merged**